pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -595,7 +595,29 @@ func ExtractEnvoyEndpoints(locEps []*LocalityEndpoints) []*endpoint.LocalityLbEn
 
 // buildEnvoyLbEndpoint packs the endpoint based on istio info.
 func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint, mtlsEnabled bool) *endpoint.LbEndpoint {
-	addr := util.BuildAddress(e.Address, e.EndpointPort)
+	addr := util.BuildAddress(e.Addresses[0], e.EndpointPort)
+
+	// This change can support multiple addresses for an endpoint, then there are some use cases for it, such as
+	// 1. An endpoint can have both ipv4 or ipv6
+	// 2. An endpoint can be represented a serviceentry/workload instance with multiple IP addresses
+	// When the additional_addresses field is populated for EDS in Envoy configuration, there would be a Happy Eyeballs
+	// algorithm to instantiate for the Endpoint, first attempt connecting to the IP address in the address field.
```

Comment:
Shouldn't we first attempt to connect to the IP family they used? It seems very odd that a user sends an IPv6 requests but then they always end up picking IPv4 (the first address).

If we are going to prefer one family vs the other, maybe it would be more appropriate to order them based on the ipFamilies order.... though I doubt that is a good idea, since K8s doesn't